Sources/ServiceKit/Container/Container.swift

```diff
@@ -1,41 +1,41 @@
-import NIO
+@_exported import protocol NIO.EventLoop
+@_exported import class NIO.EventLoopFuture
 
-/// `Container`s are used to create instances of services that your application needs in a configurable way.
-///
-///     let client = try container.make(Client.self)
-///
-/// Each `Container` has `Config`, `Environment`, and `Services`. It uses this information to dynamically provide
-/// services based on your configuration and environment.
-///
-///     switch env {
-///     case .production: config.prefer(ProductionLogger.self, for: Logger.self)
-///     default: config.prefer(DebugLogger.self, for: Logger.self)
-///     }
-///
-///     ...
-///
-///     let logger = try container.make(Logger.self) // changes based on environment
-///
-/// Containers are also `Worker`s, meaning they have a reference to an `EventLoop`.
-///
-///     print(container.eventLoop)
-///
-/// - warning: You should never use services created from a `Container` on _another_ `Container`'s `EventLoop`.
-public protocol Container: class {
+public final class Container {
+    public static func boot(env: Environment = .development, services: Services, on eventLoop: EventLoop) -> EventLoopFuture<Container> {
+        let container = Container(env: env, services: services, on: eventLoop)
+        return container.willBoot()
+            .flatMap { container.didBoot() }
+            .map { container }
+    }
+    
     /// Service `Environment` (e.g., production, dev). Use this to dynamically swap services based on environment.
-    var environment: Environment { get }
-
+    public let env: Environment
```

Comment:
Out of curiosity, any reason for this rename?

Reply:
To match [Application.env](https://github.com/vapor/vapor/blob/master/Sources/Vapor/Application.swift#L17). I've always thought `app.environment` and `Application(environment: .testing)` was very verbose. `app.env` and `Application(env: .testing)` seem nicer.